### PR TITLE
Update dockerfile to support Alpine for a slimmer base image

### PR DESCRIPTION
**Pull request by VirajP1002**
### What is the context of this PR?
To create a slimmer base image, we had to change from using `slim` to `alpine` because it built an image with a smaller virtual size than the current image. Using `slim`, the image that built had a larger virtual size than the current.  

In order for Alpine to be used, Debian-based commands (apt) had to be replaced with Alpine commands (apk). Some commands were also removed as they were considered to not be necessary. (Note:`apt-key` is also deprecated).

### How to review
- Ensure the changes made are correct
- Check if the removed commands are needed

### CODE REVIEW

**Comment by VirajP1002** [comment]
[Use a slimmer base image for eQ secrets tasks (2 SP)](https://trello.com/c/irM22AlT/5981-use-a-slimmer-base-image-for-eq-secrets-tasks-2-sp)

**Review by MebinAbraham — COMMENTED** [review]
Whats the size comparison between the two version?.

**Comment by VirajP1002** [comment]
> Whats the size comparison between the two version?.

The virtual size of the new one is 298.6 MB and the current is 458.5 MB

File: Dockerfile
Comment (liamtoozer) [inline Dockerfile]: General question - not related directly to this PR, but I couldn't spot helm being used via this image in eQ. Is this binary still needed?

File: Dockerfile
Comment (liamtoozer) [inline Dockerfile]: Really minor - it might be worth adding the `--no-cache` flag when adding packages. Might not make a difference to size but worth removing the cache after it's no longer needed? Similar to what we do in the tiny proxy Dockerfile https://github.com/ONSdigital/eq-runner-docker-tinyproxy/blob/main/Dockerfile#L7

Comment (MebinAbraham) [reply Dockerfile]: Good point. Instead of adding `--no-cache` to every package, we can run `rm /var/cache/apk/*` after installing all packages. It would be equivalent to adding `--no-cache` to every add command.

File: Dockerfile
Comment (VirajP1002) [reply Dockerfile]: I don't think it is need, I removed it from the base image and it seems to read secrets without any issues, maybe be worth anyone reviewing to test it too?

Comment (MebinAbraham) [reply Dockerfile]: This image is used for more than just the secrets, it is used as the base image for deploying benchmark which uses helm, hence we can't remove helm.

Comment (liamtoozer) [reply Dockerfile]: Thanks for confirming 👍

File: Dockerfile
Comment (MebinAbraham) [inline Dockerfile]: Instead of removing all the stuff you don't need, considering moving just the stuff you need via multi stage.

File: Dockerfile
Comment (VirajP1002) [reply Dockerfile]: Added this in to Dockerfile, thanks 👍

File: Dockerfile
Comment (VirajP1002) [reply Dockerfile]: Moved towards the multi stage since it makes the image even smaller, updated 👍

**Review by liamtoozer — APPROVED** [review]
Tested in my dev CI benchmark and it works great 👍

**Review by MebinAbraham — DISMISSED** [review]
Have not tested benchmark (wait for 1 more approval with benchmark testing)

**Review by MebinAbraham — APPROVED** [review]

**Review by petechd — APPROVED** [review]

**Pull request merged**
